### PR TITLE
Minor updates to the view-portal-object dev/troubleshooting utility s…

### PR DESCRIPTION
**Pull request by dmichaels-harvard**
* Minor updates to the view-portal-object dev/troubleshooting utility script.
* Changed from typing_extensions import Literal to import from typing;
  odd Python 3.12 issue but only in GitHub Actions (observed for submitr).
* Added tomli dependency in pyproject.toml (came up in submitr GA for Pyhthon 3.12).
* Change to structured_data.py to NOT silently convert a string representing
  a floating point number to an integer.
* Changes in structured_data.py for validator hook.

### CODE REVIEW

File: test/test_misc_utils.py [hunk 1]
```diff
@@ -3727,3 +3727,35 @@ def test_create_readonly_object():
...
+    assert to_number("4Tb", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("4TB", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("1,234,567K", allow_commas=True) is None
+    assert to_number("1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == 1234567000
+    assert to_number("-1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == -1234567000
+    assert to_number(4321) == 4321
+    assert to_number(4321.1234) == 4321.1234
+    assert to_number("1.5K", allow_multiplier_suffix=True, allow_float=True) == 1500
+    assert type(to_number("1.5K", allow_multiplier_suffix=True, allow_float=True)) == int
+    # TODO: More ...
```
Comment (willronchetti) [inline test/test_misc_utils.py:3760]: Error cases?

File: test/test_misc_utils.py [hunk 2]
```diff
@@ -3727,3 +3727,35 @@ def test_create_readonly_object():
...
+    assert to_number("3Gb", allow_multiplier_suffix=True) == 3000000000
+    assert to_number("3GB", allow_multiplier_suffix=True) == 3000000000
+    assert to_number("4T", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("4Tb", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("4TB", allow_multiplier_suffix=True) == 4000000000000
+    assert to_number("1,234,567K", allow_commas=True) is None
+    assert to_number("1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == 1234567000
+    assert to_number("-1,234,567K", allow_commas=True, allow_multiplier_suffix=True) == -1234567000
+    assert to_number(4321) == 4321
+    assert to_number(4321.1234) == 4321.1234
```
Comment (willronchetti) [inline test/test_misc_utils.py:3757]: What is the point of `allow_float` if the default behavior is to allow it? You also may want to be able to round to a specific digit. Does this test actually pass as written?

Generally I'm a little confused by `allow_float` - in my mind if passed it should always return a float, even if it is an integer, though this isn't what happens

File: dcicutils/misc_utils.py [hunk 3]
```diff
@@ -984,17 +986,109 @@ def str_to_bool(x: Optional[str]) -> Optional[bool]:
...
+
+
+def to_number(value: str,
+              allow_commas: bool = False,
+              allow_multiplier_suffix: bool = False,
+              allow_float: bool = False,
+              fallback: Optional[Union[int, float]] = None) -> Optional[Union[int, float]]:
+
+    """
+    Converts the give string value to an int, or possibly float if allow_float is True.
```
Comment (willronchetti) [inline dcicutils/misc_utils.py:1025]: `possibly float if allow_float is True`

seems to be the _default_ behavior based on your test?

**Review by willronchetti — COMMENTED** [review]
Some small comments maybe worth addressing/clarifying

File: test/test_misc_utils.py (hunk 2 above)
Comment (dmichaels-harvard) [reply test/test_misc_utils.py]: I reworked this, and tightened up tests.

File: dcicutils/misc_utils.py (hunk 3 above)
Comment (dmichaels-harvard) [reply dcicutils/misc_utils.py]: I reworked this, and tightened up tests.

File: test/test_misc_utils.py (hunk 2 above)
Comment (dmichaels-harvard) [reply test/test_misc_utils.py]: I reworked this, and tightened up tests.

File: dcicutils/misc_utils.py [hunk 4]
```diff
@@ -984,17 +986,109 @@ def str_to_bool(x: Optional[str]) -> Optional[bool]:
...
 
 
+_MULTIPLIER_K = 1000
+_MULTIPLIER_M = 1000 * _MULTIPLIER_K
+_MULTIPLIER_G = 1000 * _MULTIPLIER_M
+_MULTIPLIER_T = 1000 * _MULTIPLIER_G
+
+_MULTIPLIER_SUFFIXES = {
+    "K": _MULTIPLIER_K,
+    "KB": _MULTIPLIER_K,
```
[the remark this answers is not in the document]
Comment (dmichaels-harvard) [reply dcicutils/misc_utils.py]: Asked Sarah about this - she said case-insensitive is good, and also said to allow "b" and "bp" for "base pairs" (slightly odd I guess but okay) - lemme know if you have concerns.

**Review by willronchetti — APPROVED** [review]

**Pull request merged**